wagtail/admin/wagtail_hooks.py

```diff
@@ -933,6 +933,15 @@ def is_shown(self, request):
         return getattr(settings, "WAGTAIL_AGING_PAGES_ENABLED", True)
 
 
+class ScheduledPagesMenuItem(MenuItem):
+    def is_shown(self, request):
+        return (
+            PagePermissionPolicy()
+            .instances_user_has_permission_for(request.user, "publish")
+            .exists()
+        )
```

Comment:
We don't really care about the instances, so we should use `.user_has_permission(request.user, "publish")` which will use the cached `GroupPagePermission` objects to determine whether the user has publish permissions or not.

Using `instances_user_has_permission_for` would query the database for the instances the user can publish, and doing `.exists()` will limit the query to limit the result to 1 and return `True`, but it won't make use of the cache, hence the extra query failures in the tests.

```suggestion
        return PagePermissionPolicy().user_has_permission(request.user, "publish")
```

Reply:
Thanks for this suggestion. Actually I would suggest removing this permission check, because I have removed all publishing functionality from the report. Users will be able to see the schedules but if they want to change something they would go through normal process of doing so.